drivers/spi/spi_nxp_lpspi/spi_nxp_lpspi.c

```diff
@@ -218,12 +218,16 @@ static inline void lpspi_handle_tx_irq(const struct device *dev)
 static inline void lpspi_end_xfer(const struct device *dev)
 {
 	LPSPI_Type *base = (LPSPI_Type *)DEVICE_MMIO_NAMED_GET(dev, reg_base);
-	const struct lpspi_config *config = dev->config;
 	struct lpspi_data *data = dev->data;
 	struct spi_context *ctx = &data->ctx;
 
 	spi_context_complete(ctx, dev, 0);
+
+#if defined(NVIC_ClearPendingIRQ)
```

Comment:
is it not necessary for Cortex-R52? and if there's a way it can be abstracted using interrupt controller API instead of CMSIS calls, it should suit any supported architecture